### ISSUE TITLE
instance segmentation exported model with 0 score

### ISSUE THREAD

**Issue by dimakan** [open-edge-platform/training_extensions#1906]
MaskRCNN exported openvino model gets 0 score in evaluation.
Environment installed properly with docker by the relevant instructions.
Version 1.0.0 and 1.0.1 same behaviour.
Training works fine and eval on pth. Tested Yolox and export works fine with same setup.

Reproduced with e2e pytest. How come it's passing?
Any help is highly appreciated.

Please see the log: 
[test_instance_segmentation.log](https://github.com/openvinotoolkit/training_extensions/files/11009166/test_instance_segmentation.log)
of pytest attached.

<img width="719" alt="Screenshot 2023-03-18 at 19 06 54" src="https://user-images.githubusercontent.com/26635789/226122114-55be76c5-25db-484a-b42d-48dfe868306e.png">

pip list:
[pip_list.log](https://github.com/openvinotoolkit/training_extensions/files/11009242/pip_list.log)

Setup:

sys.platform: linux
Python: 3.8.10 (default, Nov 14 2022, 12:59:47) [GCC 9.4.0]
CUDA available: True
GPU 0: NVIDIA GeForce RTX 3090
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.7, V11.7.99
GCC: x86_64-linux-gnu-gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 1.13.1+cu117
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.7
  
TorchVision: 0.14.1+cu117
OpenCV: 4.5.5
MMCV: 1.7.0
MMCV Compiler: GCC 9.4
MMCV CUDA Compiler: 11.7
MMDetection: 2.28.1+865e0ca

**Comment by dimakan** [comment]
I run test_detection.py.
Yolox behaves normaly, but ATSS also seems to have issue with 0 score after export.
Adding pytest log, in case it's relevant

[test_detection.log](https://github.com/openvinotoolkit/training_extensions/files/11009234/test_detection.log)

**Comment by goodsong81** [comment]
@dimakan Thank you for the issue report.
The former is known issue which we are looking into. It might be some kind of compatibility issue after torch13.1 upgrade. You can quickly test it by downgrading the torch to 1.12.1.
The ATSS issue looks like different issue. Let us check.
@jaegukhyun Could you have a look?

BTW, we don't check accuracy itself in integration tests as they run on toy data sets.
We run weekly internal regression tests for performance validation.

**Comment by dimakan** [comment]
Thanks for confirming.
Tried to downgrade, but torch 1.12.1 depends on cuda 11.6.
Once such base installed, otx installation bring torch 1.13.1 back.
Probably because of tensorboard>=2.11.0 (not safe without upper limit)
Will try to remove tensorboard for the test and update.
Please let me know if there's any direction of the proper fix to try

Thanks in advance,
Dima

See bellow:

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 177.2/177.2 kB 25.6 MB/s eta 0:00:00
Requirement already satisfied: wheel>=0.26 in /usr/lib/python3/dist-packages (from tensorboard>=2.11.0->otx[full]==1.0.1) (0.34.2)
Collecting huggingface-hub
  Downloading huggingface_hub-0.13.2-py3-none-any.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.2/199.2 kB 26.5 MB/s eta 0:00:00
Collecting torch>=1.9.1
  Downloading torch-1.13.1-cp38-cp38-manylinux1_x86_64.whl (887.4 MB)
     ━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━ 418.4/887.4 MB 19.1 MB/s eta 0:00:25
ERROR: Operation cancelled by user
root@OMEN:~# pip list | grep torch
torch              1.12.1+cu116
torchvision        0.13.1+cu116

**Comment by jaegukhyun** [comment]
> I run test_detection.py. Yolox behaves normaly, but ATSS also seems to have issue with 0 score after export. Adding pytest log, in case it's relevant
> 
> [test_detection.log](https://github.com/openvinotoolkit/training_extensions/files/11009234/test_detection.log)

Thanks for reporting. Like @goodsong81 said, Instance-segmentation model have issue in exporting with torch1.13, however ATSS do not have export issue. In your log, ATSS training results show 0 accuracy, therefore ATSS export results should be 0. Since ATSS converges slow than other model, 2 epochs to train might not be enough. We will change lr for ATSS when testing for fast convergence.

And also, in torch1.12 the instance segmentation model will return proper export results

**Comment by dimakan** [comment]
**Managed to install torch1.12 with the following:**
pip install torch==1.12.1 torchvision==0.13.1 torchtext==0.13.1 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu116
pip install mmcv-full==1.7.0 -f https://download.openmmlab.com/mmcv/dist/cu116/torch1.12.1/index.html
pip install otx[full]==1.0.1 --upgrade-strategy "only-if-needed" --no-cache-dir


**Not 0, but really low score on car_tree_bug:**
2023-03-20 10:37:22,295 | INFO : OpenVINO metric evaluation completed
Performance(score: 0.008510638297872332, dashboard: (1 metric groups))

*Many warnings in the conversion:
[otx_export_eb2b.log](https://github.com/openvinotoolkit/training_extensions/files/11015814/otx_export_eb2b.log)

Will try on something more significant and update

**Comment by jaegukhyun** [comment]
> **Managed to install torch1.12 with the following:** pip install torch==1.12.1 torchvision==0.13.1 torchtext==0.13.1 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu116 pip install mmcv-full==1.7.0 -f https://download.openmmlab.com/mmcv/dist/cu116/torch1.12.1/index.html pip install otx[full]==1.0.1 --upgrade-strategy "only-if-needed" --no-cache-dir
> 
> **Not 0, but really low score on car_tree_bug:** 2023-03-20 10:37:22,295 | INFO : OpenVINO metric evaluation completed Performance(score: 0.008510638297872332, dashboard: (1 metric groups))
> 
> *Many warnings in the conversion: [otx_export_eb2b.log](https://github.com/openvinotoolkit/training_extensions/files/11015814/otx_export_eb2b.log)
> 
> Will try on something more significant and update

Could you report training score?

**Comment by dimakan** [comment]
### **With torch1.12:**

**For MaskRCNN-EfficientNetB2B on car_tree_bug:**
Train Performance(score: 0.33333333333333326, dashboard: (1 metric groups))
Export Performance(score: 0.008510638297872332, dashboard: (1 metric groups))

**For MaskRCNN-ResNet50 on car_tree_bug:**
Train Performance(score: 0.33333333333333326, dashboard: (1 metric groups))
Export Performance(score: 0.33333333333333326, dashboard: (1 metric groups))

**For MaskRCNN-EfficientNetB2B on my dataset:**
Train Performance(score: 0.7173913043478259, dashboard: (1 metric groups))
Export Performance(score: 0.7252747252747253, dashboard: (1 metric groups))

Will keep this setup for now...

**Comment by jaegukhyun** [comment]
@dimakan Can we close this issue?

**Comment by dimakan** [comment]
Ok.Thanks,DimaOn 14 Apr 2023, at 10:08, Jaeguk Hyun ***@***.***> wrote:﻿
@dimakan Can we close this issue?
